pkg/components/rook/component_test.go

```diff
@@ -0,0 +1,221 @@
+func TestConversion(t *testing.T) {
+	configHCL := `
+component "rook" {
+  enable_monitoring = true
+
+  node_selector = {
+    "storage_node_key1" = "storage_node_value1"
+    "storage_node_key2" = "storage_node_value2"
+  }
+
+  toleration {
+    key      = "toleration_key1"
+    operator = "Equal"
+    value    = "toleration_value1"
+    effect   = "NoSchedule"
+  }
+
+  toleration {
+    key      = "toleration_key2"
+    operator = "Equal"
+    value    = "toleration_value2"
+    effect   = "NoSchedule"
+  }
+
+  agent_toleration_key    = "agent_toleration_key"
+  agent_toleration_effect = "NoSchedule"
+
+  discover_toleration_key    = "discover_toleration_key"
+  discover_toleration_effect = "NoSchedule"
+
+  csi_plugin_node_selector = {
+    "node_key1" = "node_value1"
+    "node_key2" = "node_value2"
+  }
+
+  csi_plugin_toleration {
+    key = "other_node_key1"
+    operator = "Equal"
+    value = "other_node_value1"
+    effect = "NoSchedule"
+  }
+}
+`
+
+	m := renderManifest(t, configHCL)
+	if len(m) == 0 {
+		t.Fatalf("Rendered manifests shouldn't be empty")
+	}
+
+	deploy := getOperatorDeployment(t, m)
+
+	tcs := []struct {
+		Name string
+		Test func(*testing.T, map[string]string)
+	}{
+		{"MemoryResources", func(t *testing.T, m map[string]string) {
+			expected := "512Mi"
+			got := deploy.Spec.Template.Spec.Containers[0].Resources.Limits.Memory().String()
+			if expected != got {
+				t.Fatalf("expected: %s, got: %s", expected, got)
+			}
+		}},
+		{"NodeSelector", func(t *testing.T, m map[string]string) {
+			expected := map[string]string{
+				"storage_node_key1": "storage_node_value1",
+				"storage_node_key2": "storage_node_value2",
+			}
+			got := deploy.Spec.Template.Spec.NodeSelector
+
+			if !reflect.DeepEqual(expected, got) {
+				t.Fatalf("expcted: %v, got: %v", expected, got)
+			}
+		}},
+		{"Monitoring", func(t *testing.T, m map[string]string) {
+			expectedConfig := []string{
+				"rook-ceph/templates/ceph-cluster.yaml",
+				"rook-ceph/templates/ceph-osd.yaml",
+				"rook-ceph/templates/ceph-pools.yaml",
+				"rook-ceph/templates/csi-metrics-service-monitor.yaml",
+				"rook-ceph/templates/prometheus-ceph-v14-rules.yaml",
+				"rook-ceph/templates/service-monitor.yaml",
+			}
+
+			for _, ec := range expectedConfig {
+				if _, ok := m[ec]; !ok {
+					t.Fatalf("expected %q to be generated but it is not available", ec)
+				}
+			}
+		}},
+		{"OperatorTolerations", func(t *testing.T, m map[string]string) {
+			expected := []corev1.Toleration{
+				{Key: "toleration_key1", Operator: "Equal", Value: "toleration_value1", Effect: "NoSchedule"},
+				{Key: "toleration_key2", Operator: "Equal", Value: "toleration_value2", Effect: "NoSchedule"},
+			}
+			got := deploy.Spec.Template.Spec.Tolerations
+
+			if !reflect.DeepEqual(expected, got) {
+				t.Fatalf("expected: %v, got: %v", expected, got)
+			}
+		}},
+		{"EnvVars", func(t *testing.T, m map[string]string) {
+			// nolint:lll
+			subTCs := map[string]string{
```

Comment:
Looks to me like these test cases should sit in their own test function.

Reply:
I kept it here because branching this out involved lot of similar boiler plate. So kept is in here.

Reply:
We could put the boilerplate in a function and reuse it, couldn't we?

Reply:
Yes, changed the code to reuse the boilerplate. PTAL.